client/post-editor/restore-post-dialog.jsx

```diff
@@ -1,84 +1,89 @@
 /**
  * External dependencies
  */
-import React from 'react';
-import noop from 'lodash/noop';
+import React, { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+import { noop, get } from 'lodash';
 
 /**
  * Internal dependencies
  */
 import Dialog from 'components/dialog';
 import FormButton from 'components/forms/form-button';
-import utils from 'lib/posts/utils';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { getEditorPostId } from 'state/ui/editor/selectors';
+import { getEditedPostValue } from 'state/posts/selectors';
 
-export default React.createClass( {
-
-	displayName: 'EditorRestorePostDialog',
-
-	getDefaultProps() {
-		return {
-			onClose: noop,
-			onRestore: noop,
-			isAutosave: false
-		};
-	},
+class EditorRestorePostDialog extends Component {
+	static propTypes = {
+		translate: PropTypes.func,
+		onClose: PropTypes.func,
+		onRestore: PropTypes.func,
+		isAutosave: PropTypes.bool,
+		postType: PropTypes.string,
+		userCanDeletePost: PropTypes.bool,
+	}
 
-	propTypes: {
-		post: React.PropTypes.object.isRequired,
-		onClose: React.PropTypes.func,
-		onRestore: React.PropTypes.func,
-		isAutosave: React.PropTypes.bool
-	},
+	static defaultProps = {
+		onClose: noop,
+		onRestore: noop,
+		isAutosave: false,
+	}
 
-	restorePost() {
-		if ( this.props.isAutosave ) {
-			this.props.onRestore();
-		} else if ( utils.userCan( 'delete_post', this.props.post ) ) {
-			this.props.onRestore( 'draft' );
+	restorePost = () => {
+		const { isAutosave, userCanDeletePost, onRestore } = this.props;
+		if ( isAutosave ) {
+			onRestore();
+		} else if ( userCanDeletePost ) {
+			onRestore( 'draft' );
 		}
-	},
+	}
 
-	getDialogButtons() {
+	getDialogButtons = () => {
```

Comment:
Maybe we could move this to a variable in the `render` function. though, It's a personal preference maybe :)